programs/stake_api/src/stake_instruction.rs

```diff
@@ -320,15 +319,12 @@ pub fn process_instruction(
             if rest.len() < 3 {
                 return Err(InstructionError::InvalidInstructionData);
             }
-            let (to, rest) = &mut rest.split_at_mut(1);
-            let mut to = &mut to[0];
 
             me.withdraw(
                 lamports,
-                &mut to,
                 &sysvar::clock::from_keyed_account(&rest[0])?,
                 &sysvar::stake_history::from_keyed_account(&rest[1])?,
-                &rest[2..],
+                &mut rest[2..],
```

Comment:
gosh this is ugly, but I suppose there's not much choice...

Reply:
This is the best option I came up with...

Reply:
I think, given my comment below, that I want to rewrite process_instruction to parse out all the signers before calling the methods on the stake account.  This way, the functions keep their nice parameter semantics and the "authorize" check runs on pubkeys or something.

Go ahead and land this and I'll pick it up.  I've been wanting to rewrite process_instruction with slice patterns.

Reply:
Sounds good. I updated the withdraw() arg name. Look okay to merge? @rob-solana